### PR TITLE
[rp2040] I2C implementation

### PR DESCRIPTION
**Pull request by andryblack**
Synchronous implementation of I2C interface for RP2040

### CODE REVIEW

File: src/modm/platform/gpio/rp/set.hpp.in [hunk 1]
```diff
@@ -91,6 +91,11 @@ public:
 		(PortRegs<Gpios::port>::set_pue_pde(Gpios::pin, type==InputType::PullUp, type==InputType::PullDown), ...);
 	}
 
+	static void setSlew(bool fast)
```
Comment (salkinium) [inline src/modm/platform/gpio/rp/set.hpp.in]: perhaps `setSlewRate(SlewRate)` with `enum class SlewRate : uint8_t { Slow = 0, Fast = 1 }`?

File: src/modm/platform/i2c/rp/i2c_master.cpp.in [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+modm::platform::I2cMaster{{ id }}::start(I2cTransaction *transaction, ConfigurationHandler handler)
+{
+	if (!transaction)
+	{
+		return true;
+	}
+	if (not transaction->attaching())
+	{
+		transaction->detaching(modm::I2c::DetachCause::FailedToAttach);
+		// return false; // done at the end of the function
```
Comment (rleh) [inline src/modm/platform/i2c/rp/i2c_master.cpp.in:58]: Why not just return here?

**Review by rleh — APPROVED** [review]
Nice work!

I added an example using the MCP79411 real time clock I had laying around on my desk, tested it on real hardware and also rebased onto develop.
Ready to merge from me!

File: src/modm/platform/i2c/rp/i2c_master.cpp.in (hunk 2 above)
Comment (andryblack) [reply src/modm/platform/i2c/rp/i2c_master.cpp.in:58]: just source based on more complex implementation from another platforms

**Review by salkinium — APPROVED** [review]
Awesome, thanks!

**Review by rleh — APPROVED** [review]

**Pull request merged**